pkg/kubelet/dockershim/docker_container.go

```diff
@@ -307,7 +307,8 @@ func (ds *dockerService) StartContainer(_ context.Context, r *runtimeapi.StartCo
 // StopContainer stops a running container with a grace period (i.e., timeout).
 func (ds *dockerService) StopContainer(_ context.Context, r *runtimeapi.StopContainerRequest) (*runtimeapi.StopContainerResponse, error) {
 	err := ds.client.StopContainer(r.ContainerId, time.Duration(r.Timeout)*time.Second)
-	if err != nil {
+	// Do not return error if the container does not exist
+	if err != nil && !libdocker.IsContainerNotFoundError(err) {
```

Comment:
dockershim will be removed right now, maybe not accept new changes again, /cc @ehashman @SergeyKanzhelev 
Since the cri-api does not define a similar like NotFound(404) error, so I think fixed in cri interface specifications maybe a good way.

Reply:
the problem for old version using dockershim like 1.20 need be fixed also.

Reply:
The dockershim code will likely be removed as soon as 1.24 development opens, when this would be eligible for merge. We don't tend to hold deprecations in order to backport.

This will miss the final merge window for 1.20, although it may be backported to 1.21. We typically don't backport PRs that don't have an associated issue because it is hard to ascertain impact and reproduceability; can you find any other reports of this?

Reply:
I opened an issue https://github.com/kubernetes/kubernetes/issues/106734 for this pr